modules/gapi/include/opencv2/gapi/core.hpp

```diff
@@ -1757,6 +1833,71 @@ GAPI_EXPORTS GMat warpAffine(const GMat& src, const Mat& M, const Size& dsize, i
                              int borderMode = cv::BORDER_CONSTANT, const Scalar& borderValue = Scalar());
 //! @} gapi_transform
 
+/** @brief Finds centers of clusters and groups input samples around the clusters.
+
+The function kmeans implements a k-means algorithm that finds the centers of K clusters
+and groups the input samples around the clusters. As an output, \f$\texttt{bestLabels}_i\f$
+contains a 0-based cluster index for the \f$i^{th}\f$ sample.
+
+@note Function textual ID is "org.opencv.core.kmeansND"
+
+@param data Data for clustering. An array of N-Dimensional points with float coordinates is needed.
+Function can take GArray<Point2f>, GArray<Point3f> for 2D and 3D cases or GMat for any
+dimentionality
+
+@note In case of an N-dimentional points' set given, input Mat should be 2-dimensional, have
+a single row or column if there are N channels, or have N columns if there is a single channel.
+Mat should have @ref CV_32F depth
+
+@param K Number of clusters to split the set by.
+@param bestLabels Optional input integer array that can store the supposed initial cluster indices
+for every sample. Used when ( flags = #KMEANS_USE_INITIAL_LABELS ) flag is set.
+@param criteria The algorithm termination criteria, that is, the maximum number of iterations
+and/or the desired accuracy. The accuracy is specified as criteria.epsilon. As soon as each of
+the cluster centers moves by less than criteria.epsilon on some iteration, the algorithm stops.
+@param attempts Flag to specify the number of times the algorithm is executed using different
+initial labellings. The algorithm returns the labels that yield the best compactness (see the first
+function return value).
+@param flags Flag that can take values of cv::KmeansFlags
+
+@return Compactness measure that is computed as
+\f[\sum _i  \| \texttt{samples} _i -  \texttt{centers} _{ \texttt{labels} _i} \| ^2\f]
+after every attempt. The best (minimum) value is chosen and the corresponding labels and the
+compactness value are returned by the function.
+@return Integer array that stores the cluster indices for every sample.
+@return Array of the cluster centers.
+
+@note Basically, you can use only the core of the function: set the number of attempts to 1,
```

Comment:
If so, I suggest to drop this parameter.

Reply:
Why? No, it can, of course. This note is about one of possible ways to use `kmeans` when you want to control totally the labels and `kmeans` doesn't use initial randomization for them

Reply:
description reworked